api/app/controllers/spree/api/v2/platform/payment_methods_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Spree
+  module Api
+    module V2
+      module Platform
+        class PaymentMethodsController < ResourceController
+          private
+
+          def model_class
+            Spree::PaymentMethod
+          end
+
+          def spree_permitted_attributes
+            preferred_attributes = []
+
+            if action_name == 'update'
+              resource.defined_preferences.each do |preference|
+                preferred_attributes << "preferred_#{preference}".to_sym
+              end
+            end
+
+            Spree::PaymentMethod.json_api_permitted_attributes + [
+              { store_ids: [] }
```

Comment:
@MatthewKennedy and also I suppose we could add `store_ids` to `Spree::Base#json_api_permitted_attributes`